bin/et

```diff
@@ -9,25 +9,22 @@ set -e
 # Needed because if it is set, cd may print the path it changed to.
 unset CDPATH
 
-# On Mac OS, readlink -f doesn't work, so follow_links traverses the path one
-# link at a time, and then cds into the link destination and find out where it
-# ends up.
-#
-# The function is enclosed in a subshell to avoid changing the working directory
-# of the caller.
-function follow_links() (
-  cd -P "$(dirname -- "$1")"
-  file="$PWD/$(basename -- "$1")"
-  while [[ -h "$file" ]]; do
-    cd -P "$(dirname -- "$file")"
-    file="$(readlink -- "$file")"
-    cd -P "$(dirname -- "$file")"
-    file="$PWD/$(basename -- "$file")"
-  done
-  echo "$file"
-)
-
-SCRIPT_DIR=$(follow_links "$(dirname -- "${BASH_SOURCE[0]}")")
+# Returns the canonical path for its argument, with any symlinks resolved.
+function canonical_path() {
+  case "$(uname -s)" in
+    Linux)
+      readlink -f -- "$1"
+      ;;
+    Darwin)
+      realpath -q -- "$1"
+      ;;
+    *)
+      echo "The host platform is not supported by this tool"
+      exit 1
+  esac
```

Comment:
A more concise approach might be to check for `realpath` and if it's on the path then just assume it's the tool we want regardless of platform, otherwise fall back on `readlink -f`. Not sure which Linux distros (if any) bundle it though.